crates/services/compression/src/ports/block_source.rs

```diff
@@ -1,35 +1,37 @@
 use fuel_core_services::stream::BoxStream;
-
-/// Block with metadata needed for compression
-#[cfg_attr(feature = "test-helpers", derive(Default))]
-#[derive(Clone, Debug)]
-pub struct BlockWithMetadata {
-    block: fuel_core_types::blockchain::block::Block,
-    // todo: perhaps slice with lifetime? (https://github.com/FuelLabs/fuel-core/issues/2870)
-    events: Vec<fuel_core_types::services::executor::Event>,
-}
+pub(crate) use fuel_core_types::services::block_importer::SharedImportResult as BlockWithMetadata;
 
 pub(crate) type BlockHeight = u32;
 
-impl BlockWithMetadata {
-    /// Create a new block with metadata
-    pub fn new(
-        block: fuel_core_types::blockchain::block::Block,
-        events: Vec<fuel_core_types::services::executor::Event>,
-    ) -> Self {
-        Self { block, events }
+// we can't define a newtype that wraps over SharedImportResult because its `dyn Deref`
```

Comment:
I find this comment confusing.

1. How does the `dyn Deref` prevent us from wrapping it?
2. I'm not sure it's helpful when reading the comment in the code without the historic context. This assumes that someone would look at this code and think "why do we define the helpers in a sub module and not as methods on a newtype". In which scenario, anyone attempting the refactor would just hit the same issue and understand it. But for anyone not directly thinking "newtype" when they see this code, the comment itself is confusing.

I'd either skip the comment completely or do something like:
```suggestion
// Ideally we'd wrap `SharedInportResult` in a newtype to achieve (... <please elaborate here>).
// However, we can't define a newtype that wraps over SharedImportResult because its `dyn Deref`
```

Reply:
no comment, resolving